main/options/options.cc

```diff
@@ -613,6 +620,28 @@ void readOptions(Options &opts, int argc, char *argv[],
             throw EarlyReturnWithCode(1);
         }
 
+        if (raw.count("autogen-subclasses-parent")) {
+            if (!opts.print.AutogenSubclasses.enabled) {
+                logger->error("autogen-subclasses-parent must be used with -p autogen-subclasses");
+                throw EarlyReturnWithCode(1);
+            }
+            for (string parentClassName : raw["autogen-subclasses-parent"].as<vector<string>>()) {
+                opts.autogenSubclassesParents.push_back(parentClassName);
+            }
+        }
+
+        if (raw.count("autogen-subclasses-ignore") > 0) {
+            auto rawIgnorePatterns = raw["autogen-subclasses-ignore"].as<vector<string>>();
+            for (auto &p : rawIgnorePatterns) {
+                string_view pNormalized = stripTrailingSlashes(p);
+                if (p.at(0) == '/') {
+                    opts.autogenSubclassesAbsoluteIgnorePatterns.emplace_back(pNormalized);
+                } else {
+                    opts.autogenSubclassesRelativeIgnorePatterns.push_back(fmt::format("/{}", pNormalized));
```

Comment:
Inconsistent `{emplace,push}_back` usage across this file.

Reply:
Yea I noticed that when looking at the [`ignore` parsing](https://github.com/sorbet/sorbet/blob/master/main/options/options.cc#L504-L514) (which I cribbed from). At first I thought they were different b/c we pass `emplace` a string view but `push` a string. I guess it wouldn't make a difference in the latter case, right?

Reply:
In short, default to `emplace`.
Emplace takes anything that can be used to construct a string.

Reply:
[fixed](https://github.com/sorbet/sorbet/pull/1040/files#diff-d4f33382a64e83643b5258ca6f5b39b7R520)